tests/core/test_docstring_utils.py

```diff
@@ -0,0 +1,237 @@
+import pytest
+
+from ucai.core.utils.docstring_utils import DocstringInfo, parse_docstring
+
+
+def test_parse_docstring_empty():
+    """
+    Test that an empty docstring raises a ValueError.
+    """
+    with pytest.raises(ValueError, match="Docstring is empty"):
+        parse_docstring("")
+
+
+def test_parse_docstring_missing_description():
+    """
+    Test that a docstring without a description raises a ValueError.
+    """
+    docstring = """
+    Args:
+        x: The input value.
+    Returns:
+        int: The input value incremented by one.
+    """
+    with pytest.raises(ValueError, match="Function description is missing"):
+        parse_docstring(docstring)
+
+
+def test_parse_docstring_single_line_single_param():
+    """
+    Test parsing a simple docstring with a single-line description and one parameter.
+    """
```

Comment:
Do we need this comment? It almost the same as the function name.

Reply:
docstring might make sense for a complex test case, but I don't think we need docstring for simple test cases.

Reply:
Removed them all. The titles are explanatory enough